routes/net-state-files.go

```diff
@@ -0,0 +1,79 @@
+package routes
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/storj/storage/boltdb"
+)
+
+var (
+	ErrReadReq = errors.New("error reading request body")
+)
+
+type File struct {
+	DB *boltdb.Client
+}
+
+type Message struct {
+	Value string `json:"value"`
+}
+
+func NewFile(db *boltdb.Client) *File {
+	return &File{DB: db}
+}
+
+func (f *File) Put(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	givenPath := ps.ByName("path")
+
+	var msg Message
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&msg)
+	if err != nil {
+		log.Printf("err decoding response: %v", err)
+		return
+	}
+
+	file := boltdb.File{
+		Path:  givenPath,
+		Value: msg.Value,
+	}
+
+	if err := f.DB.Put(file); err != nil {
+		fmt.Println(err)
+	}
+
+	fmt.Fprintf(w, "PUT to %s\n", givenPath)
+}
+
+func (f *File) Get(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	fileKey := ps.ByName("path")
+
+	fileInfo, err := f.DB.Get([]byte(fileKey))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	fmt.Fprintf(w, "value: %s", fileInfo)
+}
+
+func (f *File) List(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	fileKeys, err := f.DB.List([]byte("files"))
+	if err != nil {
+		fmt.Println(err)
```

Comment:
technically, this isn't even logging. this is sending the error to the process's standard out file descriptor instead of the log facility. so i'd probably switch this to logging, and then also make sure to send an http error, and then make sure to return so the rest of the function doesn't run